File: WebShop_NULL/Models/ViewModels/ProfileModels/UserViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace WebShop_NULL.Models.ViewModels
{
    public class UserViewModel//Moved to F# TODO test
    {
        public int Id;
        [MaxLength(16)]
        [Required]
        public string Name { get; set; }
        [MaxLength(16)]
        [Required]
        public string Surname { get; set; }

        public string Email;
    }
}

File: sem2_FSharp/ViewModels/ProfileModels/UserViewModel.fs
namespace sem2_FSharp.ViewModels.ProfileModels

open System.ComponentModel.DataAnnotations

type UserViewModel() =
    [<DefaultValue>]
    val mutable Id: int
    
    [<MaxLength(16)>]
    [<Required>]
    [<DefaultValue>]
    val mutable private name: string
    
    [<MaxLength(16)>]
    [<Required>]
    [<DefaultValue>]
    val mutable private surname: string
    
    [<DefaultValue>]
    val mutable Email: string
    
    member public this.Name with get() = this.name
                            and set p = this.name <- p
                                    
    member public this.Surname with get() = this.surname
                                    and set p = this.surname <- p
